From 708f4ebf036d325a9f7dabde1080c09c41f4ac9f Mon Sep 17 00:00:00 2001
From: Wujian Sun <wujian.sun_1@nxp.com>
Date: Thu, 25 Sep 2025 10:08:22 +0800
Subject: [PATCH 1/2] MGS-7765: Blacklist MSAA for GPU Raster on Vivante GPUs

Vivante had poor performance when enbale MSAA, disable MSAA
for Vivante GPU.

The same workaround as other GPU vendor having the same issue.
https://codereview.chromium.org/1374043005

Upstream-Status: Pending

Change-Id: I7e562f29fb3039fe3c0e1ee36c04e7d8e8c10b37
Signed-off-by: Wujian Sun <wujian.sun_1@nxp.com>

diff -Naur chromium-140.0.7339.136_org/gpu/config/gpu_driver_bug_list.json chromium-140.0.7339.136/gpu/config/gpu_driver_bug_list.json
--- chromium-140.0.7339.136_org/gpu/config/gpu_driver_bug_list.json	2025-10-21 14:10:19.064204129 -0500
+++ chromium-140.0.7339.136/gpu/config/gpu_driver_bug_list.json	2025-10-21 14:22:20.000000000 -0500
@@ -3681,6 +3681,17 @@
       "features": [
         "disable_d3d12_h264_encoding"
       ]
+     },
+     {
+       "id": 458,
+       "description": "On Vivante GPUs MSAA performance is not acceptable for GPU rasterization",
+       "os": {
+         "type": "linux"
+       },
+       "gl_vendor": "Vivante Corporation",
+       "features": [
+         "msaa_is_slow"
+       ]
     }
   ]
 }
